Tasks/SSH/ssh.ts

```diff
@@ -0,0 +1,201 @@
+        var privateKey:string = tl.getEndpointDataParameter(sshEndpoint, 'privateKey', true); //private key is optional, password can be used for connecting
+        var hostname:string = tl.getEndpointDataParameter(sshEndpoint, 'host', false);
+        var port:string = tl.getEndpointDataParameter(sshEndpoint, 'port', true); //port is optional, will use 22 as default port if not specified
+        if(!port || port === '') {
+            tl._writeLine(tl.loc('UseDefaultPort'));
+            port = '22';
+        }
+
+        //setup the SSH connection configuration based on endpoint details
+        var sshConfig;
+        if (privateKey && privateKey !== '') {
+            tl.debug('Using private key for ssh connection.');
+            sshConfig = {
+                host: hostname,
+                port: port,
+                username: username,
+                privateKey: privateKey,
+                passphrase: password
+            }
+        } else {
+            //use password
+            tl.debug('Using username and password for ssh connection.');
+            sshConfig = {
+                host: hostname,
+                port: port,
+                username: username,
+                password: password
+            }
+        }
+
+        //read the run options
+        var runOptions : string = tl.getInput('runOptions', true);
+        var commands : string[];
+        var scriptFile : string;
+        var args : string;
+
+        if(runOptions === 'commands') {
+            commands = tl.getDelimitedInput('commands', '\n', true);
+        } else {
+            scriptFile = tl.getPathInput('scriptPath', true, true);
+            args = tl.getInput('args')
+        }
+
+        //setup the SSH connection
+        tl._writeLine(tl.loc('SettingUpSshConnection', sshConfig.username, sshConfig.host, sshConfig.port));
+        try {
+            sshClientConnection = await setupSshClientConnection(sshConfig);
+        } catch (err) {
+            tl.setResult(tl.TaskResult.Failed, tl.loc('ConnectionFailed', err));
+        }
+
+        if(sshClientConnection) {
+            //SSH connection successful
+            if (runOptions === 'commands') {
+                //run commands specified by the user
+                for (var i:number = 0; i < commands.length; i++) {
+                    tl.debug('Running command ' + commands[i] + ' on remote machine.');
+                    tl._writeLine(commands[i]);
+                    var returnCode:string = await runCommandOnRemoteMachine(sshClientConnection, commands[i]);
+                    tl.debug('Command ' + commands[i] + ' completed with return code = ' + returnCode);
+                }
+            } else if (runOptions === 'script') {
+                //run the script specified by the user
+                var scpConfig = sshConfig;
+                var remoteScript = '/home/' + sshConfig.username + '/' + path.basename(scriptFile);
```

Comment:
Can you specify $HOME or ~/ in the call to `Scp2Client.scp()`?  That could work until Windows SSH support arrives.

If you make this an input parameter, it would be nice to have both a default value like ~/ and `helpMarkDown` that gives [examples for each operating system](https://en.wikipedia.org/wiki/Home_directory?section=3#Default_home_directory_per_operating_system).